src/compas/files/gltf/gltf_mesh.py

```diff
@@ -210,7 +210,19 @@ def from_mesh(cls, context, mesh):
         :class:`~compas.files.GLTFMesh`
         """
         vertices, faces = mesh.to_vertices_and_faces()
-        return cls.from_vertices_and_faces(context, vertices, faces)
+        texture_coordinates = mesh.vertices_attribute("texture_coordinate")
+        vertex_normals = mesh.vertices_attribute("vertex_normal")
+        vertex_colors = mesh.vertices_attribute("vertex_color")
+
+        mesh_data = cls.from_vertices_and_faces(context, vertices, faces)
+        pd = mesh_data.primitive_data_list[0]
```

Comment:
Why is this done only for the first one?
Also, could it happen that the list does not have items and this throws an exception?